### PR TITLE
Add check and confirm page for cohort imports

### PR DESCRIPTION
**Pull request by thomasleese**
This updates the various pages related to cohort imports to match the designs we have for immunisation imports, specifically added a two-stage process where the user has to "Check and confirm" the records before they are recorded.

The implementation is based on the existing designs for immunisation import, but I expect these will need to change slightly when the designs for cohorts are finalised. I think once we know more about the similarities and differences between cohort and immunisation imports we could refactor these views in to shared components.

### CODE REVIEW

File: app/controllers/cohort_imports_controller.rb [hunk 1]
```diff
@@ -28,25 +27,43 @@ def create
 
     @cohort_import.process!
 
-    @cohort_import.record!
+    if @cohort_import.processed_only_exact_duplicates?
+      render :duplicates
+      return
```
Comment (tvararu) [inline app/controllers/cohort_imports_controller.rb:32]: If you want to be fancy:

```suggestion
      render :duplicates and return
```

File: app/controllers/cohort_imports_controller.rb [hunk 2]
```diff
@@ -28,25 +27,43 @@ def create
...
-  def cohort_import_params
-    params.fetch(:cohort_import, {}).permit(:csv)
+  def set_programme
+    @programme = policy_scope(Programme).active.find(params[:programme_id])
   end
 
-  def set_team
-    @team = current_user.team
+  def set_cohort_import
+    # TODO: @programme.cohort_imports.find
```
Comment (tvararu) [inline app/controllers/cohort_imports_controller.rb:58]: Follow-up?

File: app/controllers/cohort_imports_controller.rb [hunk 3]
```diff
@@ -28,25 +27,43 @@ def create
...
   end
 
-  def set_programme
-    @programme = policy_scope(Programme).find(params[:programme_id])
+  def set_patients
+    @patients = @cohort_import.patients
+  end
+
+  def cohort_import_params
+    params.fetch(:cohort_import, {}).permit(:csv)
```
Comment (tvararu) [inline app/controllers/cohort_imports_controller.rb:67]: Why can't we `.require(:cohort_import)` here? I have a similar issue on the ImmunisationImport flow and might have to do the same.

**Review by tvararu — APPROVED** [review]

File: app/controllers/cohort_imports_controller.rb (hunk 2 above)
Comment (thomasleese) [reply app/controllers/cohort_imports_controller.rb:58]: Yeah, although I'm going to wait until we've settled on the correct data model for cohorts. I think there will be a relationship between cohort imports and programmes, but not entirely sure yet.

File: app/controllers/cohort_imports_controller.rb (hunk 3 above)
Comment (thomasleese) [reply app/controllers/cohort_imports_controller.rb:67]: Good point, I think we probably can't use `require`.

**Pull request merged**